compiler/codegen/codegen.cpp

```diff
@@ -2509,9 +2578,49 @@ void makeBinary(void) {
     mysystem(command, "compiling generated source");
   }
 
-  if (fLibraryCompile && fLibraryPython) {
-    codegen_make_python_module();
+  if (gCodegenGPU == false) {
+    if (fLibraryCompile && fLibraryPython) {
+      codegen_make_python_module();
+    }
+  }
+}
+
+void codegen() {
+  if (no_codegen)
+    return;
+
+  codegenPartOne();
+
+#ifdef HAVE_LLVM
+  if (localeUsesGPU()) {
+
+    pid_t pid = fork();
+
+    if (pid == 0) {
+      // child process
+      gCodegenGPU = true;
+      codegenPartTwo();
+      gCodegenGPU = false;
+      exit(0);
+    } else {
+      // parent process
+      int status = 0;
+      while (wait(&status) != pid) {
+        // wait for child process
+      }
+    }
   }
+#endif
+
+  codegenPartTwo();
+
+}
+
+void makeBinary(void) {
```

Comment:
Should this function still exist?  Is the TODO and the question still relevant?

Reply:
Looks to me like we can remove it...

Reply:
But it's a timed pass that now will be timed with codegen. I think that's OK for now but removing it will require more changes than leaving it here and empty. Maybe it could have a TODO "remove makeBinary" comment in it.

Reply:
I think this change here broke our compilation time tracking :) 

A bit more than that, though. When compiling Arkouda with and without this PR for example, I see makeBinary times going down to 0, but codegen is not increasing by the same amplitude. Are we not measuring some part of compilation?

Reply:
I might have rushed the second part a bit. There may be something wrong in my setup

Reply:
Offline I had a chat with Sarah, and to be clear I don't think anything is broken, but this change threw me off a bit trying to understand what is going on when I look at compilation timings.

It was nice to be able to separate backend compilation time altogether from the Chapel compiler, but it is not something we should bend over backwards to support or anything. However, it would have saved me some time to know that this was happening before it happened ;)